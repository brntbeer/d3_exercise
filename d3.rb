require 'rubygems'
require 'sinatra'


get '/' do
  @title = "d3.js exercise BEGIN!"
  erb :index
end

get '/bargraph' do
  erb :bargraph
end


