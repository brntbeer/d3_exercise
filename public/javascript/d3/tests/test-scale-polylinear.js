require("./../lib/env-js/envjs/node");
require("./../d3");

var x = d3.scale.linear().domain([-1, 0, 1]).range(["red", "white", "green"]);
console.log("domain([-1, 0, 1]).range([\"red\", \"white\", \"green\"]):");
console.log("         -0.5  -> ", x(-0.5));
console.log("          0.0  -> ", x(0.0));
console.log("          0.5  -> ", x(0.5));
console.log("          1.0  -> ", x(1.0));
console.log("");
