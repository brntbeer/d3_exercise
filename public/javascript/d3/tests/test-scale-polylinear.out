domain([-1, 0, 1]).range(["red", "white", "green"]):
         -0.5  ->  rgb(255,128,128)
          0.0  ->  rgb(255,255,255)
          0.5  ->  rgb(128,192,128)
          1.0  ->  rgb(0,128,0)

